### PR TITLE
Update swinunetr template

### PR DESCRIPTION
**Pull request by heyufan1995**

### CODE REVIEW

File: auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml [hunk 1]
```diff
@@ -1,66 +1,78 @@
...
+  amp: true
+  early_stop: true
+  early_stop_delta: 0
+  early_stop_patience: 20
+  determ: false
+  input_channels: null
+  learning_rate:  0.0001
+  log_output_file: "$@bundle_root + '/model_fold' + str(@fold) + '/training.log'"
+  num_images_per_batch: 1
+  num_epochs: 1 #1600
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml:26]: This will break many tests

Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml:26]: Lots of tests use `num_epochs` to shorten the testing time. With such changes, all the tests need to use `training#epochs`. Same as all the other params

Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml:26]: Also it mean all the GitHub tutorials need to be updated to reflect such changes. This change of config file is actually a breaking change.

Comment (heyufan1995) [reply auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml]: @mingxin-zheng This template should be exactly the same with dints. I've updated the template to most recent dints template

File: auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml [hunk 2]
```diff
@@ -1,66 +1,85 @@
-amp: true
+---
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml:1]: Can we remove `---`?

File: auto3dseg/algorithm_templates/swinunetr/configs/transforms_infer.yaml [hunk 3]
```diff
@@ -3,21 +3,22 @@ transforms_infer:
...
+  - PLACEHOLDER_INTENSITY_NORMALIZATION
   - _target_: Orientationd
+    keys: "@image_key"
     axcodes: RAS
-    keys: '@image_key'
   - _target_: Spacingd
-    align_corners: true
-    keys: '@image_key'
+    keys: "@image_key"
+    pixdim: "$@transforms#resample_to_spacing"
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/configs/transforms_infer.yaml:16]: Hi @heyufan1995 , can you confirm `resample_to_spacing` is under `transforms`
I found it under the root level in `algo.py`
```
hyper_parameters.update({"resample_to_spacing": spacing})
```

File: auto3dseg/algorithm_templates/swinunetr/configs/transforms_train.yaml [hunk 4]
```diff
@@ -2,93 +2,94 @@ image_key: image
...
-    - true
-    keys:
-    - '@image_key'
-    - '@label_key'
-    mode:
-    - bilinear
-    - nearest
-    pixdim: "$@resample_to_spacing"
+    keys: ["@image_key", "@label_key"]
+    pixdim: "$@transforms#resample_to_spacing"
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/configs/transforms_train.yaml:24]: Same as the comment above

File: auto3dseg/algorithm_templates/swinunetr/configs/transforms_validate.yaml [hunk 5]
```diff
@@ -4,37 +4,22 @@ transforms_validate:
...
-    mode:
-    - bilinear
-    - nearest
-    pixdim: "$@resample_to_spacing"
-  - _target_: CastToTyped
-    keys: "@image_key"
-    dtype: "$torch.float32"
-  - PLACEHOLDER_INTENSITY_NORMALIZATION
+    keys: ["@image_key", "@label_key"]
+    pixdim: "$@transforms#resample_to_spacing"
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/configs/transforms_validate.yaml:17]: Same as above. Need confirmation about the level of `resample_to_spacing`

File: auto3dseg/algorithm_templates/swinunetr/scripts/algo.py [hunk 6]
```diff
@@ -255,16 +293,14 @@ def objective(trial):
...
-                if "out of memory" in str(e):
-                    return (
-                        float(num_images_per_batch)
-                        * float(num_sw_batch_size)
-                        * device_factor
-                    )
-                else:
-                    raise(e)
+                _ = subprocess.run(cmd.split(), check=True)
+            except BaseException:
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/scripts/algo.py:297]: Can we check if the exception has `out of memory` before printing `OOM`?

This is how it was processed before:
```
 except RuntimeError as e:
                if "out of memory" in str(e):
                    return (
                        float(num_images_per_batch)
                        * float(num_sw_batch_size)
                        * device_factor
                    )
                else:
                    raise(e)
```

File: auto3dseg/algorithm_templates/swinunetr/scripts/infer.py [hunk 7]
```diff
@@ -71,18 +78,20 @@ def __init__(self, config_file: Optional[Union[str, Sequence[str]]] = None, **ov
...
         self.device = torch.device("cuda:0")
-        torch.cuda.set_device(self.device)
 
         self.model = parser.get_parsed_content("network")
         self.model = self.model.to(self.device)
 
         pretrained_ckpt = torch.load(ckpt_name, map_location=self.device)
         self.model.load_state_dict(pretrained_ckpt)
-        print(f"[info] checkpoint {ckpt_name:s} loaded")
+        logger.debug(f"[info] checkpoint {ckpt_name:s} loaded")
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/scripts/infer.py:94]: How about removing [info]?

File: auto3dseg/algorithm_templates/swinunetr/scripts/infer.py [hunk 8]
```diff
@@ -173,10 +214,10 @@ def batch_infer(self):
 def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
     infer_instance = InferClass(config_file, **override)
     if infer_instance.fast:
-        print("[info] fast mode")
+        logger.debug("[info] fast mode")
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/swinunetr/scripts/infer.py:217]: How about
[info] fast mode -> Inference using fast mode?

File: auto3dseg/algorithm_templates/swinunetr/configs/transforms_infer.yaml (hunk 3 above)
Comment (heyufan1995) [reply auto3dseg/algorithm_templates/swinunetr/configs/transforms_infer.yaml:16]: yes it's defined twice in config
transforms:
  resample_to_spacing: "$@resample_to_spacing"
  lazy_resampling: true
resample_to_spacing: null

File: auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml (hunk 2 above)
Comment (heyufan1995) [reply auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml:1]: done

File: auto3dseg/algorithm_templates/swinunetr/scripts/algo.py (hunk 6 above)
Comment (heyufan1995) [reply auto3dseg/algorithm_templates/swinunetr/scripts/algo.py:297]: @dongyang0122

File: auto3dseg/algorithm_templates/swinunetr/scripts/infer.py (hunk 7 above)
Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/swinunetr/scripts/infer.py:94]: Or we can change it to [debug] to match the log level

File: auto3dseg/algorithm_templates/swinunetr/scripts/algo.py (hunk 6 above)
Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/swinunetr/scripts/algo.py:297]: Opened an issue here:
https://github.com/Project-MONAI/research-contributions/issues/222.

**Pull request merged**